### PR TITLE
optimize mediahistory reads

### PR DESCRIPTION
**Pull request by adipose**

### CODE REVIEW

File: src/mpc-hc/AppSettings.cpp [hunk 1]
```diff
@@ -3187,6 +3187,17 @@ bool CAppSettings::CRecentFileListWithMoreInfo::LoadMediaHistoryEntry(CStringW h
 
 void CAppSettings::CRecentFileListWithMoreInfo::ReadMediaHistory() {
     auto pApp = AfxGetMyApp();
+
+    std::wstring lastEntryCreatedCS;
```
Comment (clsid2) [inline src/mpc-hc/AppSettings.cpp]: Why not just use value `nowISOCS` in `lastEntryCreated` member? Then I can be compared directly to `LastEntryCreated` value read from profile.

Comment (adipose) [reply src/mpc-hc/AppSettings.cpp]: Since it is ISO date a string compare should work.  But I hate string compares for greater thans.

Comment (clsid2) [reply src/mpc-hc/AppSettings.cpp]: Just checking if string is equal is fine too. We only need to skip history update if value is unchanged.

Comment (adipose) [reply src/mpc-hc/AppSettings.cpp]: Yeah I thought of that.  Then I thought...why even have a timestamp?  Maybe just store your processid?

Comment (clsid2) [reply src/mpc-hc/AppSettings.cpp]: That is fine too.

**Comment by adipose** [comment]
It uses process ID now, but I added a sequence for each process as well (combined to a 64-bit int).  This way if the other process hasn't updated the mediahistory at all, we don't need to read again.  Or, if we were the last to update it, we don't update.

We only read if the other process has changed it, and we didn't already read that change.

**Comment by clsid2** [comment]
I have pushed a simpler variant using a value derived from timestamp.

**Pull request closed without merging**